### PR TITLE
Complete nodejs data for Performance

### PR DESCRIPTION
**Pull request by gilmoreorless**
#### Summary

This fills in the `null` and missing nodejs versions for these entries:

- `api.Performance.eventCounts`
- `api.Performance.measureUserAgentSpecificMemory`
- `api.PerformanceEntry.duration`
- `api.PerformanceMark.PerformanceMark`
- `api.PerformanceMark.detail`
- `api.PerformanceMeasure.detail`
- `api.PerformanceObserver.PerformanceObserver`
- `api.PerformanceObserver.worker_support`

Additionally, support for `api.Performance.toJSON` was added in Node 16.1.0, so I've updated that entry as well.

#### Test results and supporting details

* Most of the data comes from the "history" section of each API in [the documentation](https://nodejs.org/dist/latest-v16.x/docs/api/perf_hooks.html)
* Local testing was done to verify the results. The `PerformanceMark` constructor is not listed in the official documentation, but can be imported from the `perf_hooks` module from v16.0.0 onwards.
* Worker support in general was added to Node in 10.5.0, which is the version we list for other APIs' `worker_support` entries.

#### Related issues

Part of #13170

### CODE REVIEW

**Comment by gilmoreorless** [comment]
Currently blocked by #14667

**Comment by queengooborg** [comment]
I merged #14667, so this PR is ready after a quick merge of the latest `main`!

**Comment by gilmoreorless** [comment]
Thanks @queengooborg, I've rebased the branch.

**Review by queengooborg — APPROVED** [review]
LGTM, thank you!

**Pull request merged**
